### PR TITLE
framework for the canonical/natural coordinates of geometry models

### PR DESCRIPTION
**Pull request by MFraters**
@tjhei @bangerth @gassmoeller @mbweller

### CODE REVIEW

**Comment by gassmoeller** [comment]
I like it. So the current plan is to make it optional to implement these functions for geometry models, but throw an exception when somebody tries to use them with geometry models that do not implement them? Seems reasonable to me.

**Comment by bangerth** [comment]
Yes, looks good if you add the documentation for these two functions.

File: include/aspect/geometry_model/interface.h [hunk 1]
```diff
@@ -130,6 +130,18 @@ namespace aspect
         double depth(const Point<dim> &position) const = 0;
 
         /**
+         * TODO
+         */
+        virtual
```
Comment (bangerth) [inline include/aspect/geometry_model/interface.h]: documentation still missing

**Comment by MFraters** [comment]
Does everyone agree with this documentation? @anne-glerum

File: include/aspect/geometry_model/interface.h (hunk 1 above)
Comment (anne-glerum) [inline include/aspect/geometry_model/interface.h]: in 2d the coordinates for the current spheroid geometries are (radius, longitude)

File: include/aspect/geometry_model/interface.h [hunk 2]
```diff
@@ -130,6 +130,24 @@ namespace aspect
...
+         * model it  will be (radius, longitude/latitude) in 2d and (radius,
+         * longitude, latitude) in 3d.
+         */
+        virtual
+        std::array<double,dim> cartesian_to_natural_coodinates(const Point<dim> &position) const;
+
+        /**
+         * Undoes the action of cartesian_to_natural_coodinates, and turns the
+         * coordinate system which is most 'natural' to the geometry model into
+         * Cartesian coordinates
```
Comment (gassmoeller) [inline include/aspect/geometry_model/interface.h]: Missing dot at end of sentence

**Pull request merged**